lib/transform.js

```diff
@@ -4,22 +4,48 @@ var Fs = require('fs');
 
 // Declare internals
 
-var internals = {};
+var internals = {
+    fileCache: {}
+};
+
+internals.prime = function (extension, transform) {
+
+    require.extensions[extension] = function (localModule, filename) {
+
+        var src = Fs.readFileSync(filename, 'utf8');
+        var relativeFilename = filename.substr(process.cwd().length + 1);
+
+        src = (typeof transform === 'function') ? transform(src, relativeFilename) : src;
+
+        internals.fileCache[relativeFilename] = src;
 
+        localModule._compile(src, filename);
+    };
+};
 
 exports.install = function (settings) {
 
     settings.transform.forEach(function (transform) {
 
-        require.extensions[transform.ext] = function (module, filename) {
+        internals.prime(transform.ext, transform.transform);
+    });
+};
+
+exports.retrieveFile = function (path) {
```

Comment:
whats the reason to duplicate this function here and in coverage... would it be safe to have a single version used everywhere?